eox_core/pipeline.py

```diff
@@ -87,4 +88,31 @@ def check_disconnect_pipeline_enabled(backend, *args, **kwargs):
     It's recommended to place this function at the beginning of the pipeline.
     """
     if backend and backend.setting("BACKEND_OPTIONS", {}).get("disableDisconnectPipeline"):
+        LOG.exception("Disconnection pipeline is disabled, users are not allowed to disconnect.")
         raise NotAllowedToDisconnect()  # pylint: disable=raising-non-exception
+
+
+def check_user_info_matches(details, user, backend, *args, **kwargs):
+    """
+    This pipeline function checks whether the user information from the LMS matches the information
+    returned by the provider.
+
+    For example:
+        To avoid connection when the user's email from the LMS differs from the email returned by the provider,
+        add the setting:
+
+        "BACKEND_OPTIONS": { "matchUserFields":["email",...] }
+
+    It's recommended to place this step after the user is available (ie. after ensure_user_information step) and
+    before user association (ie. before user_association step)
+    """
+    if not user:
+        return
+
+    defined_fields = backend.setting("BACKEND_OPTIONS", {}).get("matchUserFields", []) if backend else []
```

Comment:
Why not something like:
```suggestion
    if backend:
        defined_fields = backend.setting("BACKEND_OPTIONS", {}).get("matchUserFields", [])
    else:
        return
```
?

Reply:
What about:

```
if not (user and backend):
    return
```

Thank you for the suggestion!

Reply:
Looks good.